### PR TITLE
(#6) Remove helpers moved to compatibility package

### LINKED ISSUE #6: (core-extension) Remove helpers available in compatibility package
As there are helpers that will be moved to the compatibility package, these should no longer be available in in this core extension package.

This issue is dependent on the completion of #3 and #5

### PR DESCRIPTION
**Pull request by AdmiringWorm**
This pull request updates the core extension package by removing the helpers that where previously available in the package, but is now available in the compatibility package instead.
As there is a dependency on this compatibility package it is not considered a breaking change.

Requires #10 to be merged and this PR to be rebased of it afterwards.

fixes #6

https://app.clickup.com/t/20540031/ENGTASKS-1186

### CODE REVIEW

**Comment by AdmiringWorm** [comment]
@vexx32 this PR is now ready for review

**Comment by AdmiringWorm** [comment]
scratch that, I rebased the wrong PR first

**Comment by AdmiringWorm** [comment]
@vexx32 and now this is actually ready 😄

File: src/chocolatey-core.extension/chocolatey-core.extension.nuspec [hunk 1]
```diff
@@ -3,15 +3,27 @@
...
-    <description>
-This package provides helper functions installed as a Chocolatey extension.
+    <description><![CDATA[This is a extension package that provides helper functions installed as a Chocolatey extension.
 These functions may be used in Chocolatey install/uninstall scripts by declaring this package a dependency in your package's nuspec.
-    </description>
+This package provides helper functions that can be used to enhance what is already available out of the box with Chocolatey CLI.
+This includes both features that are being tested for Chocolatey CLI itself, and other helpers to make the maintenance of
+Chocolatey packages easier.
+
+Helpers that was available in this package, and laten been added to Chocolatey CLI they will be moved to
```
Comment (vexx32) [inline src/chocolatey-core.extension/chocolatey-core.extension.nuspec]:
```suggestion
Helpers that were available in this package, and were later added to Chocolatey CLI, will be moved to
```

File: src/chocolatey-core.extension/chocolatey-core.extension.nuspec [hunk 2]
```diff
@@ -3,15 +3,27 @@
...
+the package [chocolatey-compatibility.extension](https://community.chocolatey.org/packages/chocolatey-compatibility.extension).
+
+Backwards compatibility is not considered for helpers available in this package, see the Notes below.
+
+## Notes
+
+- There is [a known bug](https://github.com/chocolatey-community/chocolatey-extensions/issues/11) in the function `Get-AppInstallLocation` with parameter `$AppNamePattern` which is internally used both as wildcard and regex patterns. This usually doesn't create any problems, but may do so if application contains regex symbols in the name, such as [notepad++](https://github.com/chocolatey-community/chocolatey-coreteampackages/issues/1198).
+- This package itself are not concerned of keeping backwards compatibility with Chocolatey CLI, if compatibility is of concern take a dependency on [chocolatey-compatibility.extension](https://community.chocolatey.org/packages/chocolatey-compatibility.extension) instead
```
Comment (vexx32) [inline src/chocolatey-core.extension/chocolatey-core.extension.nuspec]:
```suggestion
- This package itself is not concerned with keeping backwards compatibility with Chocolatey CLI. If compatibility is of concern, instead take a dependency on [chocolatey-compatibility.extension](https://community.chocolatey.org/packages/chocolatey-compatibility.extension).
```

**Review by vexx32 — COMMENTED** [review]
This looks great, thanks! Couple minor grammatical things we can probably fix up and we should be good to go. 🙂

File: src/chocolatey-core.extension/README.md [hunk 3]
```diff
@@ -1,6 +1,15 @@
 # chocolatey-core.extension
 
-This is the Powershell module that extends Chocolatey with new functions.
+This is a extension package that provides helper functions installed as a Chocolatey extension.
+These functions may be used in Chocolatey install/uninstall scripts by declaring this package a dependency in your package's nuspec.
+This package provides helper functions that can be used to enhance what is already available out of the box with Chocolatey CLI.
+This includes both features that are being tested for Chocolatey CLI itself, and other helpers to make the maintenance of
+Chocolatey packages easier.
+
+Helpers that was available in this package, and laten been added to Chocolatey CLI they will be moved to
```
Comment (vexx32) [inline src/chocolatey-core.extension/README.md]: Didn't see this one on first pass, sorry, looks like a copy of the the other one.

```suggestion
Helpers that were available in this package, and were later added to Chocolatey CLI, will be moved to
```

**Comment by AdmiringWorm** [comment]
@vexx32 I have updated the PR with the changes you suggested

**Review by vexx32 — APPROVED** [review]
Looks good to me, thanks for taking care of this! 💖

**Comment by AdmiringWorm** [comment]
Awesome, thanks for taking a look at the PR's @vexx32.

Now there is only the approval of the compatibility extension that is pending on CCR, then we are able to push a new version of this extension as well.

**Pull request merged**